docs: update build instructions on how to use the docker-yocto-builder (#68)
bisdn/bisdn-linux: README.md

build requires ~70 GiB of disk space. Adding additional CPU cores speeds up the
build time significantly.

## General requirements

* Disk space and RAM

We recommend a minimum of 8 GiB RAM and 150 GiB free disk space for building a single full
image.

* A functional Yocto build environment

This can be achieved by either installing and configuring everything as
documented in the section

> [Building your own build environment](#building-your-own-build-environment), or

Reviewer: ```suggestion
[Setting up your own build environment](#setting-up-your-own-build-environment), or
```

Author: reworded to "Installing"

by starting from the
[prebuild Yocto environment](https://github.com/bisdn/docker-yocto-builder) in a
docker container, as [described below](#using-the-docker-yocto-builder).

## Using the [docker-yocto-builder](https://github.com/bisdn/docker-yocto-builder)

To ease the process of "getting started building BISDN Linux with Yocto", we

> have compiled all dependencies needed by Yocto itself, as well as all additional
> ones for specifically BISDN Linux, into a docker container. This container can

Reviewer: ```suggestion
have compiled all dependencies needed by Yocto itself, as well as all additional
specific BISDN Linux dependencies, into a docker container. This container can
```

Author: i don't think using 'dependencies' two times in one sentence it good, but i will think about a better way to write this

Author: reworded

> be pulled directly from the GitHub container registry (ghcr)

Reviewer: ```suggestion
be pulled directly from the GitHub container registry ([ghcr](https://github.blog/2020-09-01-introducing-github-container-registry/))
```

Author: I don't think documentation (or an intro blog post about it) about ghcr belongs here.

Reviewer: Maybe some link to what it is. I could not actually find anything satisfying so maybe it's better to just leave it like this.

Author: Not sure what you mean here?

Author: updated

Reviewer: Good solution

[here](https://github.com/orgs/bisdn/packages?repo_name=docker-yocto-builder).
Please make sure you pull the 'ci' version of the container linked above
(or switch to the user 'builder' in the 'base' version).
To continue with the steps documented in the section
[Bootstrap build system](#bootstrap-build-system) you have to directly
exec into the container, while ideally also mounting a local path on your
file system (e.g. a new folder in your local /tmp) to the default output
directly inside of the container. This can be achieved by running:

```bash
docker run -ti -v /tmp/output:/home/builder/output ghcr.io/bisdn/docker-yocto-builder-ci:main bash
```

> ## Building your own build environment

Reviewer: ```suggestion
## Setting up your own build environment
```

Author: reworded to "Installing"

specifically, into a docker container. This container can be pulled directly
from the GitHub container registry [ghcr.io](https://ghcr.io)
[here](https://github.com/orgs/bisdn/packages?repo_name=docker-yocto-builder).
Please make sure you pull the 'ci' version of the container linked above
(or switch to the user 'builder' in the 'base' version).
To continue with the steps documented in the section
[Bootstrap build system](#bootstrap-build-system) you have to directly
exec into the container, while ideally also mounting a local path on your
file system (e.g. a new folder in your local /tmp) to the default TMPDIR
directly inside of the container. To ensure that this /tmp directory will be
writable by the builder user inside of the container, please make sure that you
either have the same UID (1000) as the builder user inside of the container, or
create the folder beforehand with more permissive access rights.
This can be achieved by running:

```bash
# create a folder to be mounted as /tmp inside of the container with very
# permissive access rights
mkdir /tmp/builder-tmp && chmod 0777 /tmp/builder-tmp
# start the docker container and mount the folder previously created as /tmp

> docker run -ti -v /tmp/builder-tmp:/tmp ghcr.io/bisdn/docker-yocto-builder-ci:main bash

Reviewer 2: Maybe add a simple "does it work" test here touch or so?
```suggestion
docker run -ti -v /tmp/builder-tmp:/tmp ghcr.io/bisdn/docker-yocto-builder-ci:main bash
# (within the container) check that /tmp is writable
[ -w /tmp ] && echo "I can write!"
```

Author: i think this makes it more complicated to understand and imho the description about a writable tmp dir above should be enough for that purpose

```

If you are planning to overwrite the TMPDIR in the conf/local.conf (`TMPDIR =
"/tmp/${MACHINE}"` by default) as mentioned in the section
[Configure target machine and cache directory](#configure-target-machine-and-cache-directory),
you should also consider to mount your local file system path to this new path

> inside of your container.

Reviewer 2: I would add a link at the end to where you are to continue afterwards, so the reader does not have to backtrack to find out where to continue, e.g. something like this:

```suggestion
inside of your container.

After you have successfully exec'd into the container, you are ready to [bootstrap the build system](#bootstrap-build-system).
```

build requires ~70 GiB of disk space. Adding additional CPU cores speeds up the
build time significantly.

## General requirements

* Disk space and RAM

We recommend a minimum of 8 GiB RAM and 150 GiB free disk space for building a single full
image.

* A functional Yocto build environment

This can be achieved by either installing and configuring everything as
documented in the section
[Installing your own build environment](#installing-your-own-build-environment), or
by starting from the

> [prebuild Yocto environment](https://github.com/bisdn/docker-yocto-builder) in a

Reviewer: ```suggestion
[prebuilt Yocto environment](https://github.com/bisdn/docker-yocto-builder) in a
```
